### PR TITLE
Fix extension disconnect functionality; 

### PR DESCRIPTION
**Pull request by wirednkod**
Must be reviewed and merged after #277

### CODE REVIEW

File: projects/burnr/src/utils/utils.ts [hunk 1]
```diff
@@ -34,12 +35,12 @@ export const downloadFile = (fileName: string, data: string, type: string): void
...
     const mnemonic = mnemonicGenerate(12);
     const pair = keyring.addFromMnemonic(mnemonic, { name: uniqueNamesGenerator(config) }, 'sr25519');
-    // const pair = keyring.addFromUri('//Charlie', { name: 'Charlie default' });
     return {
         userAddress: pair.address,
         userName: pair.meta.name as string || '____ _____',
         userSeed: mnemonic,
-        userJson: pair.toJson(),
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        userJson: pair.toJson as unknown as KeyringPair$Json,
```
Comment (wirednkod) [inline projects/burnr/src/utils/utils.ts:43]: This needed to be added cause without it build of burnr wallet could not happen

File: projects/extension/src/background/ConnectionManager.ts [hunk 2]
```diff
@@ -96,7 +83,7 @@ export class ConnectionManager extends (EventEmitter as { new(): StateEmitter })
   }
 
   /**
-   * disconnectTab disconnects all instances of {@link AppMediator} connected
+   * @summary disconnectTab disconnects all instances of {@link AppMediator} connected
```
Comment (raoulmillais) [inline projects/extension/src/background/ConnectionManager.ts:86]: @summary isn't a tsdoc tag, I think you're confusing with jsdoc. You can enable the eslint tsdoc plug in to check the syntax for you. Have a look in smoldot-test-utils or the connect packages for how to do it

**Comment by wirednkod** [comment]
> In the future can we have pull requests focused on one thing please (burnr changes don't belong with the disconnect functionality). That said this looks good now other than the minor issues pointed out

All burnr changes have moved to pr #289 . Only 1 change remains here intentionally and it is noted (see comment: https://github.com/paritytech/substrate-connect/pull/281#discussion_r646701504) in order to avoid breaking of build.

File: projects/extension/src/background/ConnectionManager.ts (hunk 2 above)
Comment (wirednkod) [reply projects/extension/src/background/ConnectionManager.ts:86]: Took the @summary from `packages/connect/src/SmoldotProvider/SmoldotProvider.ts` that was added in the tsDocs there and I assumed that it was correct. 
As agreed I will add the TsDoc configuration in `connect` and `extension` workspace for now and correct all issues that the configuration will show

File: packages/connect/src/SmoldotProvider/SmoldotProvider.ts [hunk 3]
```diff
@@ -101,8 +101,8 @@ export class SmoldotProvider implements ProviderInterface {
   healthPingerInterval = CONNECTION_STATE_PINGER_INTERVAL;
 
    /**
-   * @param {string}   chainSpec  The chainSpec for the WASM client
-   * @param {any}      sm         Optional (only used for testing) defaults to the actual smoldot module
+   * @param string -  chainSpec  The chainSpec for the WASM client
```
Comment (raoulmillais) [inline packages/connect/src/SmoldotProvider/SmoldotProvider.ts:104]:
```suggestion
   * @param chainSpec - The chainSpec for the WASM client
```

File: packages/connect/src/SmoldotProvider/SmoldotProvider.ts [hunk 4]
```diff
@@ -101,8 +101,8 @@ export class SmoldotProvider implements ProviderInterface {
...
+   * @param any    -  sm         Optional (only used for testing) defaults to the actual smoldot module
```
Comment (raoulmillais) [inline packages/connect/src/SmoldotProvider/SmoldotProvider.ts:105]: Remove type and "Optional" as these are inferred from the source by tsdoc
```suggestion
   * @param sm - (only used for testing) defaults to the actual smoldot module
```

File: packages/connect/src/SmoldotProvider/SmoldotProvider.ts [hunk 5]
```diff
@@ -292,19 +292,19 @@ export class SmoldotProvider implements ProviderInterface {
   }
 
   /**
-   * @summary Whether the node is connected or not.
-   * @return {boolean} true if connected
+   * Whether the node is connected or not.
+   * @returns boolean true if connected
```
Comment (raoulmillais) [inline packages/connect/src/SmoldotProvider/SmoldotProvider.ts:296]:
```suggestion
   * @returns true if connected
```

**Review by raoulmillais — CHANGES_REQUESTED** [review]

**Review by raoulmillais — APPROVED** [review]

**Pull request merged**